triggers/daml/AutoApproval.daml

```diff
@@ -134,6 +137,11 @@ handleApprovalRule party = do
     generateServiceCommands IssuanceService.Originate      originationRequests issuanceServices ++
     generateServiceCommands IssuanceService.CreateIssuance issuanceRequests issuanceServices
 
+  idVerificationRequests  <- queryByParty @RegulatorService.IdentityVerificationRequest (.provider)
+  regulatorServices       <- queryByParty @RegulatorService.Service              (.provider)
+  submitCommands $
+    generateServiceCommands RegulatorService.VerifyIdentity idVerificationRequests regulatorServices
+
   creditRequests          <- queryByParty @Custody.CreditAccountRequest (.provider)
```

Comment:
This builds, but it doesn't do anything when an `IdentityVerificationRequest` comes in. @brian-weir is it obvious why that might be from this code?

Reply:
It looks like you're missing `registeredTemplate @RegulatorService.Service` as part of the `registeredTemplates` in the trigger definition - meaning we won't receive events for this template and thus we can't query on it. 

Therefore, `regulatorServices` will always be an empty list and when you call `generateServiceCommands` with this empty list, this in turn calls `generateCommandsWithCid` which matches on `generateCommandsWithCid _ _ _ [] = []`. Then when `submitCommands' is called with an empty list it will match on `submitCommands [] = pure ()` and thus no side effects are executed on the ledger.

I understand that's a lot to take in, we can discuss further if you wish.

Reply:
I've been caught by this as well, especially when first starting writing triggers coming from DAZL bots - you need to add a `registeredTemplate` for both contracts you want to cause the trigger to run, _and_ any contract you need to query otherwise.

Reply:
Thanks, that was very helpful to understanding my problem and also how the trigger works in general! After registering the Service template, the Operator now auto-approves verified identity requests successfully. 

I'd still like to find a way to auto-approve the Regulator service requests themselves. Will you available after today's catch up to go through it with me?